job_dsl/rpc_gating_merge_trigger.groovy

```diff
@@ -0,0 +1,78 @@
+// Can't use env.FOO = {FOO} to transfer JJB vars to groovy
+// as this file won't be templated by JJB.
+// Alternative is to use parameters with JJB vars as the defaults.
+library "rpc-gating@${RPC_GATING_BRANCH}"
+common.globalWraps(){
+
+  // We do not want to trigger any of these
+  // for rpc-gating forks or branches other
+  // than 'master'. So if this is triggered
+  // by anything else, end now.
+  // TODO(odyssey4me):
+  // Remove once RE-1645 is resolved.
+  if (env.RPC_GATING_BRANCH != "master") {
+    print "Triggered by branch ${RPC_GATING_BRANCH} instead of master. Skipping triggers."
+    currentBuild.result = 'SUCCESS'
+    return
+  }
+
+  stage("Build-Gating-Venv") {
+    build(
+      job: "Build-Gating-Venv",
+      wait: true,
+      parameters: [
+        [
+          $class: 'StringParameterValue',
+          name: 'RPC_GATING_BRANCH',
+          value: env.RPC_GATING_BRANCH
+        ]
+      ]
+    ) // build
+  } // stage
+
+  stage("Jenkins-Job-Builder") {
+    build(
+      job: "Jenkins-Job-Builder",
+      wait: true,
+      parameters: [
+        [
+          $class: 'StringParameterValue',
+          name: 'RPC_GATING_BRANCH',
+          value: env.RPC_GATING_BRANCH
+        ]
+      ]
+    ) // build
+  } // stage
+
+  // To prevent having to continually update a list of jobs to run
+  // in parallel once the two blocking jobs are complete, we match
+  // the jobs against a regex and execute everything that matches
+  // it in parallel.
+  jobNames = common.getMatchingJobs("(?:Build-Summary-Docker-Build|Setup-.*)")
+
+  def parallelBuilds = [:]
+
+  // Cannot do for (job in jobNames), see:
+  // https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
+  for (j in jobNames) {
+    def job = j
+    parallelBuilds[job] = {
+      stage(job) {
+        build(
+          job: job,
+          wait: false, // no need to wait for these, they're non-blocking
```

Comment:
Maybe it would be better to have concurrent set to false in the merge job and wait here. In theory, it seems like two merges in quick succession could result in a situation where an older build finish second resulting in the job regressing.

Reply:
I agree, at least its parallel waiting.

Reply:
Good point. OK, given this and the suggestion about not using regex, I think I'll convert this into a list and execute with a wait.